Azure/azure-cli
# {Packaging} Bump cryptography to 38.0.4, pyOpenSSL to 22.1.0

Close #24780.
This also fixes "Failed building wheel for cryptography" error on Mac.

**Related command**
Installing packages though `azdev setup`

**Description**
This change fixes a `Could not build wheels for cryptography` error which occurs when installing packages on an M1 Mac via `azdev setup`.

**How to reproduce**
Install prerequisites:
```
pip3 install azdev
git clone https://github.com/Azure/azure-cli-extensions
cd ./azure-cli-extensions
```
Run azdev setup, which will fail when installing azure-cli darwin packages.
```
azdev setup --repo ./  --cli ../azure-cli

=======================
| Azure CLI Dev Setup |
=======================

Azure CLI:
    /Users/benbrady/Documents/azure-cli

Azure CLI extension repos:
    /Users/benbrady/Documents/azure-cli-extensions-r2k1

 Installing packages 
=====================

Upgrading pip...
Installing `requirements.txt`...
Installing `azure-cli-telemetry`...
Installing `azure-cli-core`...
Installing `azure-cli`...
Installing `azure-cli-testsdk`...
Installing `requirements.py3.Darwin.txt`...
Command `/Users/benbrady/Documents/azure-cli-extensions-r2k1/env/bin/python -m pip install -r /Users/benbrady/Documents/azure-cli/src/azure-cli/requirements.py3.Darwin.txt` failed with exit code 1:

...

      clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include -I/Users/benbrady/Documents/azure-cli-extensions-r2k1/env/include -I/Users/benbrady/.pyenv/versions/3.10.8/include/python3.10 -c build/temp.macosx-13.0-arm64-cpython-310/_openssl.c -o build/temp.macosx-13.0-arm64-cpython-310/build/temp.macosx-13.0-arm64-cpython-310/_openssl.o -Wconversion -Wno-error=sign-conversion
      build/temp.macosx-13.0-arm64-cpython-310/_openssl.c:575:10: fatal error: 'openssl/opensslv.h' file not found
      #include <openssl/opensslv.h>
               ^~~~~~~~~~~~~~~~~~~~
      1 error generated.
      
          =============================DEBUG ASSISTANCE=============================
          If you are seeing a compilation error please try the following steps to
          successfully install cryptography:
          1) Upgrade to the latest pip and try again. This will fix errors for most
             users. See: https://pip.pypa.io/en/stable/installing/#upgrading-pip
          2) Read https://cryptography.io/en/latest/installation.html for specific
             instructions for your platform.
          3) Check our frequently asked questions for more information:
             https://cryptography.io/en/latest/faq.html
          =============================DEBUG ASSISTANCE=============================
      
      error: command '/usr/bin/clang' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for cryptography
Failed to build cryptography
ERROR: Could not build wheels for cryptography, which is required to install pyproject.toml-based projects

```

**Testing Guide**
On an *M1 Mac* run the following.
```
pip3 install azdev
git clone https://github.com/Azure/azure-cli-extensions
cd ./azure-cli-extensions
azdev setup --repo ./  --cli ../azure-cli
```
Observe darwin packaged are installed successfully with no errors.
```
...
Installing `requirements.py3.Darwin.txt`...

Elapsed time: 3 min 2 sec

 Finished dev setup! 
=====================
```

This checklist is used to make sure that common guidelines for a pull request are followed.
